src/basic/fileio.c

```diff
@@ -54,6 +54,32 @@ int fdopen_unlocked(int fd, const char *options, FILE **ret) {
         return 0;
 }
 
+int take_fdopen_unlocked(int *fd, const char *options, FILE **ret) {
+        int     r;
+
+        assert(fd);
+
+        r = fdopen_unlocked(*fd, options, ret);
+        if (r < 0)
+                return r;
+
+        *fd = -1;
+
+        return 0;
+}
+
+FILE * take_fdopen(int *fd, const char *options) {
+        assert(fd);
+
+        FILE *f = fdopen(*fd, options);
+        if (!f)
+                return NULL;
```

Comment:
hmm, if we do a wrapper like this, maybe use our usual way for propagating errors here, too? i.e. like fdopen_unlocked() is not a 1:1 wrapper around fdopen() either, it changes the return handling to follow our own usual style (i.e. kernel-style negative errno returns), so we should do the same for this wrapper too?

Reply:
Eh, why go mucking with the call sites more than necessary?  It's not like the wrapper is making multiple calls that clobber errno or anything to guard against, so I'm inclined to keep it this way and if we feel like pursuing consistent-ifying everything in the future then we can go rearrange call sites more.

Reply:
ok, then let's merge as soon as CI passes. I figure we can just leave this PR open and close the other one?

Reply:
Sure